src/extension/background-script/actions/nostr/setPrivateKey.ts

```diff
@@ -1,10 +1,32 @@
+import { encryptData } from "~/common/lib/crypto";
 import type { MessagePrivateKeySet } from "~/types";
 
 import state from "../../state";
 
 const setPrivateKey = async (message: MessagePrivateKeySet) => {
-  await state.getState().getNostr().setPrivateKey(message.args.privateKey);
-  return {};
+  const id = message.args?.id || state.getState().currentAccountId;
+
+  const password = state.getState().password as string;
+  const privateKey = message.args.privateKey;
+  const accounts = state.getState().accounts;
+
+  if (id && Object.keys(accounts).includes(id)) {
```

Comment:
Just a nitpick: wouldn't it be enough to have `if (Object.keys(accounts).includes(id))` in the condition? In case `id` is `undefined` the condition is also `false`. Or do I miss something?

Reply:
Yeah, it should work, but TypeScript (🙂) requires the `includes()` parameter to strictly be `string` and not `null` type